pkg/resource/deploy/plan_executor.go

```diff
@@ -213,6 +195,55 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	return res
 }
 
+func (pe *planExecutor) performDeletes(ctx context.Context, opts Options) result.Result {
+	defer func() {
+		// We're done here - signal completion so that the step executor knows to terminate.
+		pe.stepExec.SignalCompletion()
+	}()
+
+	prev := pe.plan.prev
+	if prev == nil || len(prev.Resources) == 0 {
+		return nil
+	}
+
+	logging.V(7).Infof("performDeletes(...): beginning")
+
+	deleteSteps, res := pe.stepGen.GenerateDeletes(opts.DestroyTargets)
+	if res != nil {
+		logging.V(7).Infof("performDeletes(...): generating deletes produced error result")
+		return res
+	}
+
+	deletes := pe.stepGen.ScheduleDeletes(deleteSteps)
+
+	resourceToStep := make(map[*resource.State]Step)
```

Comment:
Can we sink this into the `if len(opts.DestroyTargets)` block below?

Reply:
done.  thanks.